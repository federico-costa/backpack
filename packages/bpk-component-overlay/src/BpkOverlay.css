@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-overlay__wrapper{position:relative}.bpk-overlay__overlay{position:absolute;top:0;left:0;width:100%;height:100%}.bpk-overlay__overlay--tint{background:rgba(2,18,44,0.56)}.bpk-overlay__overlay--border-radius-sm{border-radius:.375rem}
