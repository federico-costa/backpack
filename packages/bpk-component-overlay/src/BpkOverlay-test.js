/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import React from 'react';
import renderer from 'react-test-renderer';

import BpkOverlay, { BORDER_RADIUS_STYLES, OVERLAY_TYPES } from './BpkOverlay';

describe('BpkOverlay', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <BpkOverlay>
          <span>Backpack</span>
        </BpkOverlay>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with foregroundContent', () => {
    const tree = renderer
      .create(
        <BpkOverlay foregroundContent={<span>Skyscanner</span>}>
          <span>Backpack</span>
        </BpkOverlay>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  Object.keys(OVERLAY_TYPES).map(overlayType =>
    it(`should render correctly with overlayType={${overlayType}}`, () => {
      const tree = renderer
        .create(
          <BpkOverlay overlayType={overlayType}>
            <span>Backpack</span>
          </BpkOverlay>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    }),
  );

  Object.keys(BORDER_RADIUS_STYLES).map(borderRadiusStyle =>
    it(`should render correctly with borderRadiusStyle={${borderRadiusStyle}}`, () => {
      const tree = renderer
        .create(
          <BpkOverlay borderRadiusStyle={borderRadiusStyle}>
            <span>Backpack</span>
          </BpkOverlay>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    }),
  );

  it('should support custom class names', () => {
    const tree = renderer
      .create(
        <BpkOverlay className="custom-classname">
          <span>Backpack</span>
        </BpkOverlay>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support arbitrary props', () => {
    const tree = renderer
      .create(
        <BpkOverlay testID="123">
          <span>Backpack</span>
        </BpkOverlay>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
