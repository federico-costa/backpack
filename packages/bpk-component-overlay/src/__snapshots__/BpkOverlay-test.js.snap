// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkOverlay should render correctly 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  />
</div>
`;

exports[`BpkOverlay should render correctly with borderRadiusStyle={none} 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  />
</div>
`;

exports[`BpkOverlay should render correctly with borderRadiusStyle={sm} 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--border-radius-sm bpk-overlay__overlay--tint"
  />
</div>
`;

exports[`BpkOverlay should render correctly with foregroundContent 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  >
    <span>
      Skyscanner
    </span>
  </div>
</div>
`;

exports[`BpkOverlay should render correctly with overlayType={none} 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay"
  />
</div>
`;

exports[`BpkOverlay should render correctly with overlayType={tint} 1`] = `
<div
  className="bpk-overlay__wrapper"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  />
</div>
`;

exports[`BpkOverlay should support arbitrary props 1`] = `
<div
  className="bpk-overlay__wrapper"
  testID="123"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  />
</div>
`;

exports[`BpkOverlay should support custom class names 1`] = `
<div
  className="bpk-overlay__wrapper custom-classname"
>
  <span>
    Backpack
  </span>
  <div
    className="bpk-overlay__overlay bpk-overlay__overlay--tint"
  />
</div>
`;
